### PR TITLE
#173: Deprecating endpoint mapping in favor of plain middleware

### PR DESCRIPTION
**Pull request by ch-ti8m-michalpenka**
see #173

### CODE REVIEW

**Comment by m-wild** [comment]
@VisualBean @yurvon-screamo are either of you familiar with aspnetcore? I haven't kept up to date with the recent changes to endpoints vs middleware.

File: src/Saunter/AsyncApiMiddleware.cs [hunk 1]
```diff
@@ -3,57 +3,57 @@
...
-        private bool IsRequestingAsyncApiSchema(HttpRequest request)
-        {
-            return HttpMethods.IsGet(request.Method) && request.Path.IsMatchingRoute(_options.Middleware.Route);
-        }
+    private static async Task RespondWithAsyncApiSchemaJsonAsync(
+        HttpResponse response,
+        AsyncApiSchema.v2.AsyncApiDocument asyncApiSchema,
+        IAsyncApiDocumentSerializer asyncApiDocumentSerializer)
+    {
+        var asyncApiSchemaJson = asyncApiDocumentSerializer.Serialize(asyncApiSchema);
```
Comment (VisualBean) [inline src/Saunter/AsyncApiMiddleware.cs:53]: This could be cached instead of serialising every time

**Comment by VisualBean** [comment]
> @VisualBean @yurvon-screamo are either of you familiar with aspnetcore? I haven't kept up to date with the recent changes to endpoints vs middleware.

Middleware structure looks fine. 
My only comments on this is around caching of values. I'll take a deeper look Monday though

File: src/Saunter/AsyncApiMiddleware.cs (hunk 1 above)
Comment (ch-ti8m-michalpenka) [reply src/Saunter/AsyncApiMiddleware.cs:53]: My apologies but that's what's been used all the time. That's not what this PR is about. It's about enabling others to use plain middlewares in order to be less limited.

That being said, thanks for your review.

**Comment by yurvon-screamo** [comment]
I disagree with internal sealed middleware. 

Overloading or single usage are real scenarios.

**Comment by thompson-tomo** [comment]
I agree with @VisualBean that the project would benefit from caching but let's track that via: https://github.com/asyncapi/saunter/issues/198

**Comment by yurvon-screamo** [comment]
due to the lack of understanding of the benefits to the project from this change and any activities related to it, I close the request

**Comment by yurvon-screamo** [comment]
due to the lack of understanding of the benefits to the project from this change and any activities related to it, I close the request

**Pull request closed without merging**